lib/ecto/query/builder.ex

```diff
@@ -218,6 +219,12 @@ defmodule Ecto.Query.Builder do
     error! "`#{Macro.to_string(other)}` is not a valid query expression"
   end
 
+  def runtime_fragment!([{_,_}|_] = kw) when is_list(kw), do: kw
+  def runtime_fragment!(expr) do
+    raise ArgumentError, "to prevent sql injection, only a keyword list may be interpolated " <>
```

Comment:
That's what we use in other places, so I think we are fine: https://github.com/parkerl/ecto/blob/fragment_interpolation_validation/lib/ecto/query/builder.ex#L434